librz/core/cmd/cmd_debug.c

```diff
@@ -690,20 +674,106 @@ static int step_line(RzCore *core, int times) {
 				continue;
 			}
 			rz_core_reg_update_flags(core);
-			eprintf("Cannot retrieve dwarf info at 0x%08" PFMT64x "\n", off);
+			RZ_LOG_ERROR("cannot retrieve dwarf info at 0x%08" PFMT64x "\n", off);
 			return false;
 		}
 	} while (!strcmp(file, file2) && line == line2);
 
-	eprintf("--> 0x%08" PFMT64x " %s : %d\n", off, file2, line2);
+	RZ_LOG_INFO("--> 0x%08" PFMT64x " %s : %d\n", off, file2, line2);
 	tmp_ptr = rz_file_slurp_line(file2, line2, 0);
-	eprintf("--> %s\n", tmp_ptr);
+	RZ_LOG_INFO("--> %s\n", tmp_ptr);
 	free(tmp_ptr);
 	rz_core_reg_update_flags(core);
+	return true;
+}
 
+/**
+ * \brief Step back
+ * \param core The RzCore instance
+ * \param steps Step steps
+ * \return success
+ */
+RZ_API bool rz_core_debug_step_back(RzCore *core, int steps) {
+	if (!rz_core_is_debug(core)) {
+		if (!rz_core_esil_step_back(core)) {
+			RZ_LOG_ERROR("cannot step back\n");
+			return false;
+		}
+	} else {
+		if (!core->dbg->session) {
+			RZ_LOG_ERROR("session has not started\n");
+			return false;
+		}
+		if (rz_debug_step_back(core->dbg, steps) < 0) {
+			RZ_LOG_ERROR("stepping back failed\n");
+			return false;
+		}
+		rz_core_reg_update_flags(core);
+	}
 	return true;
 }
 
+/**
+ * \brief Step over
+ * \param core The RzCore instance
+ * \param steps Step steps
+ */
+RZ_API void rz_core_debug_step_over(RzCore *core, int steps) {
+	if (rz_config_get_i(core->config, "dbg.skipover")) {
+		rz_core_debug_step_skip(core, steps);
+	} else {
+		if (rz_core_is_debug(core)) {
```

Comment:
Same here, invert the condition, add return after the steps loop with ESIL, then the following `else` is unnecessary.